x-pack/legacy/plugins/apm/typings/elasticsearch.ts

```diff
@@ -71,6 +72,10 @@ declare module 'elasticsearch' {
     >;
   };
 
+  type SamplerAggregation<SubAggregationMap> = SubAggregation<
```

Comment:
It's equivalent to `type SamplerAggregation<T> = SubAggregation<T>;` - the extra name is not necessary but makes this line easier later, instead of `{ sampler: SubAggregation<AggregationOption[AggregationName]>; }`

Reply:
These types are a bit raw, it was our first attempt of getting aggregation inference. I'm actually super impressed that you've managed to make it work 😅